compiler/qsc_project/src/project.rs

```diff
@@ -135,54 +174,49 @@ pub trait FileSystem {
 
     /// Given a path, list its directory contents (if any).
     fn list_directory(&self, path: &Path) -> miette::Result<Vec<Self::Entry>>;
-    /// Given an initial path, fetch files matching <`initial_path`>/**/*.qs
-    fn collect_project_sources(&self, initial_path: &Path) -> miette::Result<Vec<Self::Entry>> {
-        let listing = self.list_directory(initial_path)?;
-        if let Some(src_dir) = listing.into_iter().find(|x| {
-            let Ok(entry_type) = x.entry_type() else {
-                return false;
-            };
-            entry_type == EntryType::Folder && x.entry_name() == "src"
-        }) {
-            self.collect_project_sources_inner(&src_dir.path())
-        } else {
-            Err(miette::ErrReport::msg(
-                "No `src` directory found for project.",
-            ))
-        }
-    }
 
-    fn collect_project_sources_inner(
-        &self,
-        initial_path: &Path,
-    ) -> miette::Result<Vec<Self::Entry>> {
-        let listing = self.list_directory(initial_path)?;
-        let mut files = vec![];
-        for item in filter_hidden_files(listing.into_iter()) {
-            match item.entry_type() {
-                Ok(EntryType::File) if item.entry_extension() == "qs" => files.push(item),
-                Ok(EntryType::Folder) => {
-                    files.append(&mut self.collect_project_sources_inner(&item.path())?);
-                }
-                _ => (),
-            }
-        }
-        Ok(files)
-    }
+    fn resolve_path(&self, base: &Path, path: &Path) -> miette::Result<PathBuf>;
 
     /// Given a [`ManifestDescriptor`], load project sources.
     fn load_project(&self, manifest: &ManifestDescriptor) -> miette::Result<Project> {
-        let project_path = manifest.manifest_dir.clone();
-        let qs_files = self.collect_project_sources(&project_path)?;
-
-        let qs_files = qs_files.into_iter().map(|file| file.path());
+        // Rather than rewriting all the async code in the project loader,
+        // we call the async implementation here, doing some tricks to make it
+        // run synchronously.
+
+        let fs = ToFileSystemAsync { fs: self };
+
+        // WARNING: This will panic if there are *any* await points in the
+        // load_project implementation. Right now, we know that will never be the case
+        // because we just passed in our synchronous FS functions to the project loader.
+        // Proceed with caution if you make the `FileSystemAsync` implementation any
+        // more complex.
+        FutureExt::now_or_never(fs.load_project(manifest)).expect("load_project should never await")
+    }
```

Comment:
Looking for @sezna 's blessing and/or comments on this crate in particular - and especially this method. This was the only way I could find to combine the sync/async implementations, but happy to discuss if this is concerning